src/ui/widgets.rs

```diff
@@ -0,0 +1,345 @@
+    ui.heading("Counter Demo");
+    ui.add_space(10.0);
+
+    ui.horizontal(|ui| {
+        if ui.button("➖ Decrement").clicked() {
+            *counter -= 1;
+            info!("Counter decremented to {}", counter);
+        }
+
+        ui.add_space(10.0);
+        ui.label(format!("Value: {}", counter));
+        ui.add_space(10.0);
+
+        if ui.button("➕ Increment").clicked() {
+            *counter += 1;
+            info!("Counter incremented to {}", counter);
+        }
+
+        ui.add_space(10.0);
+
+        if ui.button("🔄 Reset").clicked() {
+            *counter = 0;
+            info!("Counter reset");
+        }
+    });
+
+    ui.add_space(20.0);
+    ui.separator();
+    ui.add_space(20.0);
+}
+
+pub fn render_audio_section(ui: &mut egui::Ui, is_playing: bool) -> AudioAction {
+    ui.heading("Audio Test");
+    ui.add_space(10.0);
+
+    let mut action = AudioAction::None;
+
+    ui.horizontal(|ui| {
+        let button_text = if is_playing {
+            "🔊 Playing..."
+        } else {
+            "🔔 Play Beep"
+        };
+
+        if ui
+            .add_enabled(!is_playing, egui::Button::new(button_text))
+            .clicked()
+        {
+            action = AudioAction::Play;
+        }
+
+        ui.add_space(10.0);
+
+        if ui
+            .add_enabled(is_playing, egui::Button::new("⏹ Stop"))
+            .clicked()
+        {
+            action = AudioAction::Stop;
+        }
+    });
+
+    ui.add_space(20.0);
+    ui.separator();
+    ui.add_space(20.0);
+
+    action
+}
+
+pub fn render_info_section(ui: &mut egui::Ui) {
+    ui.group(|ui| {
+        ui.label("ℹ️ About this demo:");
+        ui.add_space(5.0);
+        ui.label("• Edit text in the input field");
+        ui.label("• Use counter buttons to increment/decrement");
+        ui.label("• Play a test beep sound (440Hz A4 note)");
+        ui.label("• Close the window to exit the application");
+    });
+}
+
+pub fn render_library_section(
+    ui: &mut egui::Ui,
+    library: &[Song],
+    library_path: Option<&Path>,
+    filter: &mut String,
+    theme: Theme,
+) -> LibraryAction {
+    // Section header
+    ui.horizontal(|ui| {
+        ui.colored_label(theme.primary(), "■");
+        ui.heading(
+            egui::RichText::new("LIBRARY DATABASE")
+                .color(theme.text_primary())
+                .strong(),
+        );
+    });
+    ui.add_space(10.0);
+
+    let mut action = LibraryAction::None;
+
+    // Library status and scan button
+    ui.horizontal(|ui| {
+        if let Some(path) = library_path {
+            ui.label(
+                egui::RichText::new(format!("PATH: {}", path.display()))
+                    .color(theme.text_muted())
+                    .small(),
+            );
+            ui.add_space(5.0);
+            ui.colored_label(theme.active(), format!("[{} TRACKS]", library.len()));
+        } else {
+            ui.label(
+                egui::RichText::new("[NO DATABASE LOADED]")
```

Comment:
That string would be clearer about the error
```suggestion
                egui::RichText::new("Library folder doesn't exist")
```